hogql_parser/parser.cpp

```diff
@@ -291,6 +291,447 @@ class HogQLParseTreeConverter : public HogQLParserBaseVisitor {
+    }
+
+    auto func_stmt_ctx = ctx->funcStmt();
+    if (func_stmt_ctx) {
+      return visit(func_stmt_ctx);
+    }
+
+    auto var_assignment_ctx = ctx->varAssignment();
+    if (var_assignment_ctx) {
+      return visit(var_assignment_ctx);
+    }
+
+    auto block_ctx = ctx->block();
+    if (block_ctx) {
+      return visit(block_ctx);
+    }
+
+    auto expr_stmt_ctx = ctx->exprStmt();
+    if (expr_stmt_ctx) {
+      return visit(expr_stmt_ctx);
+    }
+
+    auto empty_stmt_ctx = ctx->emptyStmt();
+    if (empty_stmt_ctx) {
+      return visit(empty_stmt_ctx);
+    }
+
+    throw ParsingError("Statement must be one of returnStmt, ifStmt, whileStmt, forStmt, funcStmt, varAssignment, "
+                       "block, exprStmt, or emptyStmt");
+  }
+
+  VISIT(ExprStmt) {
+    PyObject* expr;
+    try {
+      expr = visitAsPyObject(ctx->expression());
+    } catch (...) {
+      throw;
+    }
+    PyObject* ret = build_ast_node("ExprStatement", "{s:N}", "expr", expr);
+    if (!ret) {
+      Py_DECREF(expr);
+      throw PyInternalError();
+    }
+    return ret;
+  }
+
+  VISIT(ReturnStmt) {
+    PyObject* expr;
+    try {
+      expr = visitAsPyObjectOrNone(ctx->expression());
+    } catch (...) {
+      throw;
+    }
+    PyObject* ret = build_ast_node("ReturnStatement", "{s:N}", "expr", expr);
+    if (!ret) {
+      Py_DECREF(expr);
+      throw PyInternalError();
+    }
+    return ret;
+  }
+
+  VISIT(IfStmt) {
+    PyObject* expr;
+    try {
+      expr = visitAsPyObject(ctx->expression());
+    } catch (...) {
+      throw;
+    }
+    PyObject* then_stmt;
+    try {
+      then_stmt = visitAsPyObject(ctx->statement(0));
+    } catch (...) {
+      Py_DECREF(expr);
+      throw;
+    }
+    PyObject* else_stmt;
+    try {
+      else_stmt = visitAsPyObjectOrNone(ctx->statement(1));
+    } catch (...) {
+      Py_DECREF(expr);
+      Py_DECREF(then_stmt);
+      throw;
+    }
+    PyObject* ret = build_ast_node("IfStatement", "{s:N,s:N,s:N}", "expr", expr, "then", then_stmt, "else_", else_stmt);
+    if (!ret) {
+      Py_DECREF(expr);
+      Py_DECREF(then_stmt);
+      Py_DECREF(else_stmt);
+      throw PyInternalError();
+    }
+    return ret;
+  }
+
+  VISIT(WhileStmt) {
+    PyObject* expr;
+    try {
+      expr = visitAsPyObject(ctx->expression());
+    } catch (...) {
+      throw;
+    }
+    PyObject* body;
+    try {
+      body = visitAsPyObjectOrNone(ctx->statement());
+    } catch (...) {
+      Py_DECREF(expr);
+      throw;
+    }
+    PyObject* ret = build_ast_node("WhileStatement", "{s:N,s:N}", "expr", expr, "body", body);
+    if (!ret) {
+      Py_DECREF(expr);
+      Py_DECREF(body);
+      throw PyInternalError();
+    }
+    return ret;
+  }
+
+  VISIT(ForStmt) {
+    PyObject* initializer;
+    auto initializer_var_declr_ctx = ctx->initializerVarDeclr;
+    auto initializer_var_assignment_ctx = ctx->initializerVarAssignment;
+    auto initializer_expression_ctx = ctx->initializerExpression;
```

Comment:
Given this is just member access, there'd be no difference in performance to avoid the declarations and use e.g. `ctx->initializerVarDeclr` directly in this method